openhands/core/config/utils.py

```diff
@@ -191,16 +190,15 @@ def load_from_toml(cfg: AppConfig, toml_file: str = 'config.toml'):
 
 def finalize_config(cfg: AppConfig):
     """More tweaks to the config after it's been loaded."""
-    cfg.workspace_base = os.path.abspath(cfg.workspace_base)
-    # Set workspace_mount_path if not set by the user
-    if cfg.workspace_mount_path is UndefinedString.UNDEFINED:
-        cfg.workspace_mount_path = cfg.workspace_base
-
-    if cfg.workspace_mount_rewrite:  # and not config.workspace_mount_path:
-        # TODO why do we need to check if workspace_mount_path is None?
-        base = cfg.workspace_base or os.getcwd()
-        parts = cfg.workspace_mount_rewrite.split(':')
-        cfg.workspace_mount_path = base.replace(parts[0], parts[1])
+    if cfg.workspace_base is not None:
```

Comment:
This is a nice clean up, just to note quickly though, we may need to check that it works with evals. I seem to recall that unfortunately, during the lifetime of the `UndefinedString`, we actually starting using None as a valid value somewhere... I want to say it has the meaning that workspace is not mounted.

Oh or maybe I'm missing some change here 🤔

Reply:
For example, this relies on None: https://github.com/All-Hands-AI/OpenHands/blob/2a6740f4ba904801090def5508d30ec1e432ced7/evaluation/swe_bench/eval_infer.py#L88

Reply:
Good call out! Will ask Xingyao to test

Reply:
I think the reason why we have UNDEFINED is because we default workspace_base to certain value. But now they are default to None, I think it is no longer an issue!

Reply:
Forgot to mention--I did a little eval here and everything worked!